doc/designs/disable-stale-users.md

```diff
@@ -0,0 +1,147 @@
+# Disable Stale Users
+
+**DESIGN STAGE**
+
+## Overview
+
+Some regulations call for accounts that haven't been used for a specific
+amount of time be automatically disabled in the identity management system.
+
+FreeIPA does not keep track of successful authentication events.
+The krbLastSuccessfulAuth attribute is not updated by default since
+https://pagure.io/freeipa/issue/5313 for reliability reasons.
+That attribute was not replicated either.
+Therefore distinguishing between stale and active accounts cannot be done with
+existing logon data.
+
+However, user passwords have to be changed regularly in these environments as
+well. A user who failed to change his password in time can be considered stale
+and therefore be disabled by the system.
+In this case, a grace period should be provided between password expiration and
+the actual account locking event.
+
+The DisableStaleUsers (DSU) tooling is provided to make sure no non-locked-out
+account has a password expired since N days, N being 7 by default.
+Installing the DSU tooling will not change the behavior of existing clusters.
+DSU has to be explicitely configured and enabled to start disabling stale
+accounts.
+
+Accounts are locked out by setting the nsAccountLock attribute
+(OID: 2.16.840.1.113730.3.1.610) to True.
+
+On very busy clusters, using a non-busy replica (like a hidden replica) for DSU
+is recommended.
+
+
+## Limitations
+
+If regulations call for stale/inactive accounts to be disabled within X
+number of days, then the password lifetime of the passwords for these
+accounts must be set to a value lower than X, minus the expected grace period
+mentioned above, minus the maximum amount of time between two invocations
+of the DSU tool.
+
+
+## Example
+
+Given the following requirements and facts:
+* stale accounts must be disabled after 90 days' inactivity
+* grace period is decided to be 8 days
+* the tool runs every day
+
+setting the maximum password lifetime to 80 days or less would meet the
+guidelines as 80+8+1 is less than 90.
+
+## Multiple runners
+
+DSU can be installed on more than a single replica. This provides redundancy.
+However, care must be taken that DSU is not triggered both:
+* at the same time
+* on the same accounts
```

Comment:
What is the risk of running at the same time? At worst it would disable the same account at the same time right?

Reply:
nsAccountLock is not present until a user account has been locked once. I'd rather not have two ADD operations on the same entry + attribute at the same time, though the end value is going to be the same.

Reply:
As you suggested, I've added a randomized timer which should be enough to alleviate issues, plus there would not be any collision anyway.